src/main/java/org/myrobotlab/service/Arduino.java

```diff
@@ -1651,17 +1576,18 @@ public BoardInfo publishBoardInfo(Integer version/* byte */,
       broadcastState();
     }
 
-    if (boardInfo != null) {
-      DeviceSummary[] ds = boardInfo.getDeviceSummary();
-      if (deviceList.size() - 1 > ds.length) { /* -1 for self */
-        sync();
-      }
-    }
+    // TODO: consider, can we really just re-sync when we see begin only.. ?  feels better/safer.
+//    if (boardInfo != null) {
+//      DeviceSummary[] ds = boardInfo.getDeviceSummary();
+//      if (deviceList.size() - 1 > ds.length) { /* -1 for self */
+//        log.info("Invoking Sync DeviceList: {} and DeviceSummary: {}", deviceList, ds);
+//        invoke("sync");
+//      }
+//    }

```

Comment:
does device sync still work ?

Reply:
sync works now any time a MrlCommBegin message is seen.  at that point.. a "send" call is created to sync() ..  the actual re-attaching of the devices needs to be done from a different thread so it doesn't cause a deadlock waiting on an ack in the Msg.java class.   

you can't initiate a sync call in the publishBoardInfo because there might be a slight lag between the current deviceList in arduino vs the device list being published in the MrlComm board info...